src/sinks/influxdb/logs.rs

```diff
@@ -217,6 +212,30 @@ impl HttpSink for InfluxDbLogsSink {
 }
 
 impl InfluxDbLogsConfig {
+    fn get_measurement(&self) -> String {
+        let measurement = match self.measurement.clone() {
+            Some(measure) => match self.namespace.clone() {
+                Some(_) => {
+                    warn!("Option `namespace` has been preceded by `measurement`.");
+                    measure
+                }
+                None => measure,
+            },
+            None => match self.namespace.clone() {
+                Some(namespace) => {
+                    warn!(
+                        "Option `namespace` has been deprecated. Use `measurement` instead. \
+                           For example, you can use `measurement=<namespace>.vector` for the \
+                           same effect."
+                    );
+                    format!("{}.vector", namespace)
+                }
+                None => panic!("Option `measurement` is required."),
+            },
+        };
+        measurement
```

Comment:
While this is not performance critical, this structure can be simplified quite a bit by matching on both `measurement` and `namespace` at once and using references instead of clones for the namespace. Also, this function must not panic. Return a `Result` here and return that error in the caller above.
```suggestion
        match (self.measurement.as_ref(), self.namespace.as_ref()) {
            (Some(measure), Some(_)) => {
                warn!("Option `namespace` has been preceded by `measurement`.");
                measure.clone()
            }
            (Some(measure), None)) => measure.clone(),
            (None, Some(namespace)) => {
                warn!(
                    "Option `namespace` has been deprecated. Use `measurement` instead. \
                       For example, you can use `measurement=<namespace>.vector` for the \
                       same effect."
                );
                format!("{}.vector", namespace)
            }
            (None, None) => panic!("Option `measurement` is required."),
        }
```

Reply:
I was looking for the "matching both at once", and just did not know how to do that. I'll do that.